Alright, I think this is good to go since none of the build failures seem to be related to this update. Some notes on the build failures below.


On `aarch64-linux`, the `llama-index` packages are failing due to dependency `python3Packages.chromadb` failing during `pythonImportsCheckPhase` (CC @fabaff):

```
chromadb-aarch64-linux> Executing pythonImportsCheckPhase
chromadb-aarch64-linux> Check whether the following modules can be imported: chromadb
chromadb-aarch64-linux> Error in cpuinfo: failed to parse the list of possible processors in /sys/devices/system/cpu/possible
chromadb-aarch64-linux> Error in cpuinfo: failed to parse the list of present processors in /sys/devices/system/cpu/present
chromadb-aarch64-linux> Error in cpuinfo: failed to parse both lists of possible and present processors
chromadb-aarch64-linux> terminate called after throwing an instance of 'onnxruntime::OnnxRuntimeException'
chromadb-aarch64-linux>   what():  /build/source/include/onnxruntime/core/common/logging/logging.h:294 static const onnxruntime::logging::Logger& onnxruntime::logging::LoggingManager::DefaultLogger() Attempt to use DefaultLogger but none has been registered.
```

On `x86_darwin`, the `checkov` package is failing to build due to dependency `python311Packages.igraph` failing 3 tests all with the following error (CC @dotlambda):

```
Error at src/linalg/arpack.c:1098: ARPACK error. -- No shifts could be applied during a cycle of the Implicitly restarted Arnoldi iteration. One possibility is to increase the size of NCV relative to NEV
```

On `{aarch64,x86_64}-darwin`,  the `llama-{index,parse}` packages are failing during `pythonImportsCheckPhase` (CC @fabaff):

```
llama-parse> [nltk_data] Downloading package stopwords to /nix/store/b683x2v11g6812
llama-parse> [nltk_data]     i3hxxl7c0fb15pn399-python3.11-llama-index-
llama-parse> [nltk_data]     core-0.10.29/lib/python3.11/site-
llama-parse> [nltk_data]     packages/llama_index/core/_static/nltk_cache...
llama-parse> Traceback (most recent call last):
llama-parse>   File "/nix/store/b683x2v11g6812i3hxxl7c0fb15pn399-python3.11-llama-index-core-0.10.29/lib/python3.11/site-packages/llama_index/core/utils.py", line 60, in __init__
llama-parse>     nltk.data.find("corpora/stopwords", paths=[self._nltk_data_dir])
llama-parse>   File "/nix/store/4zyjip8iymigqs3vwjin5iw3wy21xl0i-python3.11-nltk-3.8.1/lib/python3.11/site-packages/nltk/data.py", line 583, in find
llama-parse>     raise LookupError(resource_not_found)
llama-parse> LookupError: 
llama-parse> **********************************************************************
llama-parse>   Resource stopwords not found.
llama-parse>   Please use the NLTK Downloader to obtain the resource:
llama-parse> 
llama-parse>   >>> import nltk
llama-parse>   >>> nltk.download('stopwords')
llama-parse>   
llama-parse>   For more information see: https://www.nltk.org/data.html
llama-parse> 
llama-parse>   Attempted to load corpora/stopwords
llama-parse> 
llama-parse>   Searched in:
llama-parse>     - '/nix/store/b683x2v11g6812i3hxxl7c0fb15pn399-python3.11-llama-index-core-0.10.29/lib/python3.11/site-packages/llama_index/core/_static/nltk_cache'
llama-parse> **********************************************************************
llama-parse> 
llama-parse> 
llama-parse> During handling of the above exception, another exception occurred:
llama-parse> 
llama-parse> Traceback (most recent call last):
llama-parse>   File "<string>", line 1, in <module>
llama-parse>   File "<string>", line 1, in <lambda>
llama-parse>   File "/nix/store/j0jaqv09421wydqmfawcnwvv2f32q86a-python3-3.11.8/lib/python3.11/importlib/__init__.py", line 126, in import_module
llama-parse>     return _bootstrap._gcd_import(name[level:], package, level)
llama-parse>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
llama-parse>   File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
llama-parse>   File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
llama-parse>   File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
llama-parse>   File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
llama-parse>   File "<frozen importlib._bootstrap_external>", line 940, in exec_module
llama-parse>   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
llama-parse>   File "/nix/store/mm8i4yhj4xhwxw45pbz3v1f7dd8d6m9z-python3.11-llama-parse-0.4.0/lib/python3.11/site-packages/llama_parse/__init__.py", line 1, in <module>
llama-parse>     from llama_parse.base import LlamaParse, ResultType
llama-parse>   File "/nix/store/mm8i4yhj4xhwxw45pbz3v1f7dd8d6m9z-python3.11-llama-parse-0.4.0/lib/python3.11/site-packages/llama_parse/base.py", line 10, in <module>
llama-parse>     from llama_index.core.async_utils import run_jobs
llama-parse>   File "/nix/store/b683x2v11g6812i3hxxl7c0fb15pn399-python3.11-llama-index-core-0.10.29/lib/python3.11/site-packages/llama_index/core/__init__.py", line 10, in <module>
llama-parse>     from llama_index.core.base.response.schema import Response
llama-parse>   File "/nix/store/b683x2v11g6812i3hxxl7c0fb15pn399-python3.11-llama-index-core-0.10.29/lib/python3.11/site-packages/llama_index/core/base/response/schema.py", line 8, in <module>
llama-parse>     from llama_index.core.schema import NodeWithScore
llama-parse>   File "/nix/store/b683x2v11g6812i3hxxl7c0fb15pn399-python3.11-llama-index-core-0.10.29/lib/python3.11/site-packages/llama_index/core/schema.py", line 15, in <module>
llama-parse>     from llama_index.core.utils import SAMPLE_TEXT, truncate_text
llama-parse>   File "/nix/store/b683x2v11g6812i3hxxl7c0fb15pn399-python3.11-llama-index-core-0.10.29/lib/python3.11/site-packages/llama_index/core/utils.py", line 89, in <module>
llama-parse>     globals_helper = GlobalsHelper()
llama-parse>                      ^^^^^^^^^^^^^^^
llama-parse>   File "/nix/store/b683x2v11g6812i3hxxl7c0fb15pn399-python3.11-llama-index-core-0.10.29/lib/python3.11/site-packages/llama_index/core/utils.py", line 62, in __init__
llama-parse>     nltk.download("stopwords", download_dir=self._nltk_data_dir)
llama-parse>   File "/nix/store/4zyjip8iymigqs3vwjin5iw3wy21xl0i-python3.11-nltk-3.8.1/lib/python3.11/site-packages/nltk/downloader.py", line 777, in download
llama-parse>     for msg in self.incr_download(info_or_id, download_dir, force):
llama-parse>   File "/nix/store/4zyjip8iymigqs3vwjin5iw3wy21xl0i-python3.11-nltk-3.8.1/lib/python3.11/site-packages/nltk/downloader.py", line 642, in incr_download
llama-parse>     yield from self._download_package(info, download_dir, force)
llama-parse>   File "/nix/store/4zyjip8iymigqs3vwjin5iw3wy21xl0i-python3.11-nltk-3.8.1/lib/python3.11/site-packages/nltk/downloader.py", line 701, in _download_package
llama-parse>     os.makedirs(os.path.join(download_dir, info.subdir))
llama-parse>   File "<frozen os>", line 225, in makedirs
llama-parse> PermissionError: [Errno 13] Permission denied: '/nix/store/b683x2v11g6812i3hxxl7c0fb15pn399-python3.11-llama-index-core-0.10.29/lib/python3.11/site-packages/llama_index/core/_static/nltk_cache/corpora'
```

--- a/pkgs/development/python-modules/openai/default.nix
+++ b/pkgs/development/python-modules/openai/default.nix
@@ -25,7 +25,7 @@
 
 buildPythonPackage rec {
   pname = "openai";
-  version = "1.17.0";
+  version = "1.20.0";
   pyproject = true;
 
   disabled = pythonOlder "3.7.1";
@@ -34,7 +34,7 @@ buildPythonPackage rec {
     owner = "openai";
     repo = "openai-python";
     rev = "refs/tags/v${version}";
-    hash = "sha256-NAfG8edQB9L72MlESqMXw6fo/s2F/BjAf5yRAmx+J/s=";
+    hash = "sha256-aR/OEOz6xUKjsZk3lynx0SZJ4lnWk0uDFioO/NakVl8=";
   };
 
   build-system = [

--- a/pkgs/development/python-modules/pulsar-client/default.nix
+++ b/pkgs/development/python-modules/pulsar-client/default.nix
@@ -93,12 +93,12 @@ let
       "3.12-aarch64-darwin" = getSrcFromPypi {
         platform = "macosx_10_15_universal2";
         dist = "cp312";
-        hash = "sha256-9/jw/wr1oUD9pOadVAaMRL081iVMUXwVgnUMcG1UNvE=";
+        hash = "sha256-Hgd6SDm+Pq094/BbTCRCadyi3wf0fOoLkFRMfp3BZC8=";
       };
       "3.12-x86_64-darwin" = getSrcFromPypi {
         platform = "macosx_10_15_universal2";
         dist = "cp312";
-        hash = "sha256-9/jw/wr1oUD9pOadVAaMRL081iVMUXwVgnUMcG1UNvE=";
+        hash = "sha256-Hgd6SDm+Pq094/BbTCRCadyi3wf0fOoLkFRMfp3BZC8=";
       };
     };
 

